src/adaptations/device-layer/ESP32/BLEManagerImpl.cpp

```diff
@@ -604,16 +599,16 @@ WEAVE_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
         ExitNow();
     }
 
-    // Construct the Weave Service Data to be sent in the scan response packet.
+    // Construct the Weave BLE Service Data to be sent in the scan response packet.  On the ESP32,
+    // the buffer given to esp_ble_gap_config_adv_data() must begin with the Weave BLE service UUID.
     memcpy(weaveServiceData.ServiceUUID, ShortUUID_WoBLEService, sizeof(weaveServiceData.ServiceUUID));
-    weaveServiceData.DataBlockLen = 16;
-    weaveServiceData.DataBlockType = 0x16;
-    weaveServiceData.DataBlockMajorVersion = 0;
-    weaveServiceData.DataBlockMinorVersion = 1;
-    Encoding::LittleEndian::Put16(weaveServiceData.DeviceVendorId, (uint16_t)WEAVE_DEVICE_CONFIG_DEVICE_VENDOR_ID);
-    Encoding::LittleEndian::Put16(weaveServiceData.DeviceProductId, (uint16_t)WEAVE_DEVICE_CONFIG_DEVICE_PRODUCT_ID);
-    Encoding::LittleEndian::Put64(weaveServiceData.DeviceId, FabricState.LocalNodeId);
-    weaveServiceData.PairingStatus = ConfigurationMgr().IsPairedToAccount() ? 1 : 0;
+    weaveServiceData.DeviceIdBlock.Init();
+    weaveServiceData.DeviceIdBlock.SetVendorId(WEAVE_DEVICE_CONFIG_DEVICE_VENDOR_ID);
+    weaveServiceData.DeviceIdBlock.SetProductId(WEAVE_DEVICE_CONFIG_DEVICE_PRODUCT_ID);
+    weaveServiceData.DeviceIdBlock.SetDeviceId(FabricState.LocalNodeId);
+    weaveServiceData.DeviceIdBlock.PairingStatus = ConfigurationMgr().IsPairedToAccount()
```

Comment:
I don't want to make WeaveBLEDeviceIdentificationBlock dependent on Device Layer code, because the Device Layer isn't built for all OpenWeave builds.  E.g., this would preclude the type from being used in client applications to parse received beacons.

However, I do like the idea of consolidating this code into a common helper function.  I think I would place this function on the Device Layer's ConfigurationManager object, e.g. ConfigurationManager::SetBLEDeviceIdData().

Reply:
I pushed an updated change that incorporates the helper function.

Reply:
That works. I was initially thinking a SetPairingStatus(Bool isPairedToAccount).